src/net_processing.cpp

```diff
@@ -153,7 +154,12 @@ namespace {
     std::atomic<int64_t> g_last_tip_update(0);
 
     /** Relay map */
-    typedef std::map<uint256, CTransactionRef> MapRelay;
+    struct RelayEntry {
+        explicit RelayEntry(CTransactionRef &&tx) : m_txref(tx) {}
```

Comment:
Use `m_txref(std::move(tx))` here. `tx` in this context is _not_ an rvalue reference (as it's been bound to variable), it's only initialized by accepting an rvalue reference.

Reply:
I had no idea, thank you for pointing this out!